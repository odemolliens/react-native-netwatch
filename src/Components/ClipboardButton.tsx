import * as React from 'react';
import { useContext } from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { ThemeContext } from '../Theme';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

export interface IProps {
  onPress: () => void;
}

// Button use in the details screens to copy to clipboard a text. It's just a button text.
export const ClipboardButton: React.FC<IProps> = (props: IProps) => {
  const theme = useContext(ThemeContext);
  return (
    <TouchableOpacity testID="buttonCopyToClipboard" style={styles.container} onPress={props.onPress}>
      <MaterialCommunityIcons name="clipboard-arrow-left-outline" color={theme.primaryColor} size={14} />
      <Text style={[{ color: theme.primaryColor, marginLeft: 6 }]}>Copy</Text>
    </TouchableOpacity>
  );
};

export default ClipboardButton;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderLeftWidth: 0,
    alignItems: 'center',
  },
});
