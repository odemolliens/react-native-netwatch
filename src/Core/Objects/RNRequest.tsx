import { ILog, LogType } from '../../types';
// @ts-ignore
// BlobFileReader is needed to read the content of the body response (it's type can be blob)
import BlobFileReader from 'react-native/Libraries/Blob/FileReader';

const fromEntries = (arr: any[]) =>
  arr.reduce((acc, [k, v]) => {
    acc[k] = v;
    return acc;
  }, {});

export const stringifyData = (data: any) => {
  try {
    if (data?._parts?.length) {
      return JSON.stringify(fromEntries(data?._parts), null, 2);
    }
    return JSON.stringify(JSON.parse(data), null, 2);
  } catch (e) {
    return `${data}`;
  }
};

export const getRequestBody = (dataSent: any): string => stringifyData(dataSent) || '';

export const getResponseBody = async (responseType: string, response?: any): Promise<string> => {
  if (!response) return '';
  const _responseBody = await (responseType !== 'blob' ? response : parseResponseBlob(response));
  return stringifyData(_responseBody || '');
};

export const parseResponseBlob = async (response: Blob) => {
  const blobReader = new BlobFileReader();
  blobReader.readAsText(response);

  return await new Promise<string>((resolve, reject) => {
    const handleError = () => reject(blobReader.error);

    blobReader.addEventListener('load', () => {
      resolve(blobReader.result);
    });
    blobReader.addEventListener('error', handleError);
    blobReader.addEventListener('abort', handleError);
  });
};

export class RNRequest implements ILog {
  _id: number = -1;
  type: LogType = 'RNR';
  startTime: number = Date.now();
  readyState: number = 0;
  url: string = '';
  shortUrl?: string = '';
  method: string = '';
  status: number = -1;
  endTime: number = 0;
  timeout?: number;
  dataSent: string = '';
  requestHeaders?: any;
  responseHeaders?: any;
  responseContentType?: string;
  responseSize?: number;
  responseType?: string;
  responseURL?: string;
  response: string = '';

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default RNRequest;
