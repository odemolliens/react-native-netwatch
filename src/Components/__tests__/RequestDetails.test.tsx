import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { RequestDetails, IProps } from '../RequestDetails';
import { RNRequest } from '../../Core/Objects/RNRequest';
import { NRequest } from '../../Core/Objects/NRequest';
import Clipboard from '@react-native-clipboard/clipboard';
import ClipboardButton from '../ClipboardButton';

jest.mock('@react-native-clipboard/clipboard', () => ({
  setString: jest.fn(),
}));

describe('Status test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;
  const onPressViewMoreRequest = jest.fn();
  const onPressViewMoreResponse = jest.fn();
  const setSnackBarMessage = jest.fn();
  const setSnackBarVisibility = jest.fn();

  it('should render request', () => {
    givenProps(mockRequest);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly with request with image', () => {
    givenProps(mockRequestwithImage);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly with request with image', () => {
    givenProps(mockRequestwithImageError);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should copy to clipboard', () => {
    givenProps(mockRequest);
    givenComponent();
    component.find(ClipboardButton).first().simulate('press');
    expect(Clipboard.setString).toHaveBeenCalledTimes(1);
  });

  it('should render NRequest properly', () => {
    givenProps(mockRequestwithFailed);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly with request with headers', () => {
    givenProps(mockRequestwithWarning);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<RequestDetails {...props} />);
  }

  function givenProps(item: RNRequest) {
    props = {
      item,
      onPressViewMoreRequest,
      onPressViewMoreResponse,
      setSnackBarMessage,
      setSnackBarVisibility,
    };
  }

  const mockRequest: RNRequest = new RNRequest({
    _id: 73,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'NR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  });

  const mockRequestwithImage: NRequest = new NRequest({
    _id: 76,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL:
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'NR',
    url:
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==',
  });

  const mockRequestwithImageError: NRequest = new NRequest({
    _id: 76,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'data:image/png;base64,',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'NR',
    url: 'data:image/png;-,',
  });

  const mockRequestwithWarning: NRequest = new NRequest({
    _id: 76,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 301,
    timeout: 0,
    type: 'NR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  });

  const mockRequestwithFailed: NRequest = new NRequest({
    _id: 76,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 500,
    timeout: 0,
    type: 'NR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  });
});
