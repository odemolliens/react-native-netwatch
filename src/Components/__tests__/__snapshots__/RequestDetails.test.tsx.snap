// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status test suite should render NRequest properly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Title
      style={
        Array [
          Object {
            "marginRight": 6,
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      GET
    </Title>
    <Component
      style={
        Array [
          Object {
            "borderRadius": 4,
            "fontSize": 12,
            "height": 22,
            "justifyContent": "center",
            "width": 30,
          },
          Object {
            "backgroundColor": "#B91C1C",
          },
        ]
      }
      testID="statusCodeColor"
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#F9FAFB",
            },
          ]
        }
      >
        500
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Hostname : 
    </Text>
    <Text>
      run.mocky.io
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Started at : 
    </Text>
    <Text>
      16/02/2021 - 12:12:54   ( 1015ms )
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Timeout : 
    </Text>
    <Text>
      0
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Info
  </Subheading>
  <Component
    key="0"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6B7280",
            },
          ]
        }
      >
        url
      </Text>
      <ClipboardButton
        onPress={[Function]}
      />
    </Component>
    <Text
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Text>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6B7280",
            },
          ]
        }
      >
        responseURL
      </Text>
    </Component>
    <Text
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Data
  </Subheading>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      dataSent
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Response Body
  </Subheading>
  <Component
    style={
      Object {
        "flexDirection": "row-reverse",
        "paddingHorizontal": 16,
      }
    }
  >
    <ClipboardButton
      onPress={[Function]}
    />
  </Component>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      response
    </Text>
  </Component>
</Component>
`;

exports[`Status test suite should render properly with request with headers 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Title
      style={
        Array [
          Object {
            "marginRight": 6,
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      GET
    </Title>
    <Component
      style={
        Array [
          Object {
            "borderRadius": 4,
            "fontSize": 12,
            "height": 22,
            "justifyContent": "center",
            "width": 30,
          },
          Object {
            "backgroundColor": "#D97706",
          },
        ]
      }
      testID="statusCodeColor"
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#F9FAFB",
            },
          ]
        }
      >
        301
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Hostname : 
    </Text>
    <Text>
      run.mocky.io
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Started at : 
    </Text>
    <Text>
      16/02/2021 - 12:12:54   ( 1015ms )
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Timeout : 
    </Text>
    <Text>
      0
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Info
  </Subheading>
  <Component
    key="0"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6B7280",
            },
          ]
        }
      >
        url
      </Text>
      <ClipboardButton
        onPress={[Function]}
      />
    </Component>
    <Text
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Text>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6B7280",
            },
          ]
        }
      >
        responseURL
      </Text>
    </Component>
    <Text
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Data
  </Subheading>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      dataSent
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Response Body
  </Subheading>
  <Component
    style={
      Object {
        "flexDirection": "row-reverse",
        "paddingHorizontal": 16,
      }
    }
  >
    <ClipboardButton
      onPress={[Function]}
    />
  </Component>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      response
    </Text>
  </Component>
</Component>
`;

exports[`Status test suite should render properly with request with image 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Title
      style={
        Array [
          Object {
            "marginRight": 6,
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      GET
    </Title>
    <Component
      style={
        Array [
          Object {
            "borderRadius": 4,
            "fontSize": 12,
            "height": 22,
            "justifyContent": "center",
            "width": 30,
          },
          Object {
            "backgroundColor": "#4D7C0F",
          },
        ]
      }
      testID="statusCodeColor"
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#F9FAFB",
            },
          ]
        }
      >
        200
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Hostname : 
    </Text>
    <Text>
      image
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Started at : 
    </Text>
    <Text>
      16/02/2021 - 12:12:54   ( 1015ms )
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Timeout : 
    </Text>
    <Text>
      0
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Info
  </Subheading>
  <Component
    key="0"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "minHeight": 100,
        }
      }
    >
      <Component
        onError={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "80%",
          }
        }
      />
    </Component>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "minHeight": 100,
        }
      }
    >
      <Component
        onError={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "80%",
          }
        }
      />
    </Component>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Data
  </Subheading>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      dataSent
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Response Body
  </Subheading>
  <Component
    style={
      Object {
        "flexDirection": "row-reverse",
        "paddingHorizontal": 16,
      }
    }
  >
    <ClipboardButton
      onPress={[Function]}
    />
  </Component>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      response
    </Text>
  </Component>
</Component>
`;

exports[`Status test suite should render properly with request with image 2`] = `
<Component
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Title
      style={
        Array [
          Object {
            "marginRight": 6,
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      GET
    </Title>
    <Component
      style={
        Array [
          Object {
            "borderRadius": 4,
            "fontSize": 12,
            "height": 22,
            "justifyContent": "center",
            "width": 30,
          },
          Object {
            "backgroundColor": "#4D7C0F",
          },
        ]
      }
      testID="statusCodeColor"
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#F9FAFB",
            },
          ]
        }
      >
        200
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Hostname : 
    </Text>
    <Text>
      image
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Started at : 
    </Text>
    <Text>
      16/02/2021 - 12:12:54   ( 1015ms )
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Timeout : 
    </Text>
    <Text>
      0
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Info
  </Subheading>
  <Component
    key="0"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "minHeight": 100,
        }
      }
    >
      <Component
        onError={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": "data:image/png;-,",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "80%",
          }
        }
      />
    </Component>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "minHeight": 100,
        }
      }
    >
      <Component
        onError={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": "data:image/png;base64,",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "80%",
          }
        }
      />
    </Component>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Data
  </Subheading>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      dataSent
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Response Body
  </Subheading>
  <Component
    style={
      Object {
        "flexDirection": "row-reverse",
        "paddingHorizontal": 16,
      }
    }
  >
    <ClipboardButton
      onPress={[Function]}
    />
  </Component>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      response
    </Text>
  </Component>
</Component>
`;

exports[`Status test suite should render request 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Title
      style={
        Array [
          Object {
            "marginRight": 6,
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      GET
    </Title>
    <Component
      style={
        Array [
          Object {
            "borderRadius": 4,
            "fontSize": 12,
            "height": 22,
            "justifyContent": "center",
            "width": 30,
          },
          Object {
            "backgroundColor": "#4D7C0F",
          },
        ]
      }
      testID="statusCodeColor"
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#F9FAFB",
            },
          ]
        }
      >
        200
      </Text>
    </Component>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Hostname : 
    </Text>
    <Text>
      run.mocky.io
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Started at : 
    </Text>
    <Text>
      16/02/2021 - 12:12:54   ( 1015ms )
    </Text>
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#6B7280",
        }
      }
    >
      Timeout : 
    </Text>
    <Text>
      0
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Info
  </Subheading>
  <Component
    key="0"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6B7280",
            },
          ]
        }
      >
        url
      </Text>
      <ClipboardButton
        onPress={[Function]}
      />
    </Component>
    <Text
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Text>
  </Component>
  <Component
    key="1"
    style={
      Object {
        "marginBottom": 10,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6B7280",
            },
          ]
        }
      >
        responseURL
      </Text>
    </Component>
    <Text
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "color": "#F9FAFB",
          },
        ]
      }
    >
      https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Request Data
  </Subheading>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      dataSent
    </Text>
  </Component>
  <Subheading
    style={
      Array [
        Object {
          "marginBottom": 10,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#374151",
          "color": "#F9FAFB",
        },
      ]
    }
  >
    Response Body
  </Subheading>
  <Component
    style={
      Object {
        "flexDirection": "row-reverse",
        "paddingHorizontal": 16,
      }
    }
  >
    <ClipboardButton
      onPress={[Function]}
    />
  </Component>
  <Component
    style={
      Object {
        "justifyContent": "flex-start",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text>
      response
    </Text>
  </Component>
</Component>
`;
